### PR TITLE
LC GitLab integration.

### PR DESCRIPTION
**Pull request by corbett5**
This adds a LC GitLab CI that builds the TPL's via spack and then builds and tests LvArray for all eight combinations of debug/release, gcc/clang and quartz/lassen. Ideally it would also run the benchmarks for release builds and log the results. As is there are a few problems.

Problems:
- The CI is triggered manually by someone from the LLNL organization with 2FA enabled commenting "LGTM" on the PR. This triggers a run on the most recent commit.
- Even after commenting it can take some time for the GitLab mirror to pick up the changes and run the tests up to 30 minutes.
- You can only view the detailed results if you have an LC account.

### CODE REVIEW

**Comment by corbett5** [comment]
LGTM

**Comment by corbett5** [comment]
LGTM

**Comment by corbett5** [comment]
LGTM

**Comment by corbett5** [comment]
In this last case the pipeline ran about 40 minutes after the "LGTM".

**Comment by corbett5** [comment]
LGTM

**Comment by corbett5** [comment]
LGTM

**Comment by corbett5** [comment]
LGTM

**Comment by corbett5** [comment]
LGTM

File: scripts/uberenv/project.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "package_final_phase" : "hostconfig",
   "package_source_dir" : "../..",
   "spack_url": "https://github.com/corbett5/spack",
-  "spack_branch": "feature/corbett/lvarray",
+  "spack_branch": "package/corbett/geosx",
```
Comment (rrsettgast) [inline scripts/uberenv/project.json:7]: is this intentional?

**Review by rrsettgast — APPROVED** [review]
looks pretty good.

File: scripts/uberenv/project.json (hunk 1 above)
Comment (corbett5) [reply scripts/uberenv/project.json:7]: Yup. It doesn't change much.

**Pull request merged**